cli/unstable_progress_bar.ts

```diff
@@ -0,0 +1,189 @@
+// Copyright 2018-2025 the Deno authors. MIT license.
+
+/**
+ * The options that are provided to a {@link createProgressBar} or
+ * {@link ProgressBarStream}.
+ */
+export interface ProgressBarOptions {
+  /**
+   * The offset size of the input if progress is resuming part way through.
+   * @default {0}
+   */
+  currentSize?: number;
+  /**
+   * The total size expected to receive.
+   */
+  totalSize: number;
+  /**
+   * The length that the progress bar should be, in characters.
+   * @default {50}
+   */
+  barLength?: number;
+  /**
+   * The character to fill the progress bar up with as it makes progress.
+   * @default {'#'}
+   */
+  fullChar?: string;
+  /**
+   * The character the progress bar starts out with.
+   * @default {'-'}
+   */
+  emptyChar?: string;
+  /**
+   * Whether the progress bar should be removed after completion.
+   * @default {false}
+   */
+  clearBar?: boolean;
+  /**
+   * Whether or not a timer `[mm:ss]` should be displayed.
+   * @default {true}
+   */
+  includeTime?: boolean;
+  /**
+   * Whether or not a percentage `[0.00%]` should be displayed.
+   * @default {false}
+   */
+  includePercent?: boolean;
+  /**
+   * Whether or not the amount `[0.00/97.04 KiB]` passed through should be
+   * displayed.
+   * @default {true}
+   */
+  includeAmount?: boolean;
+  /**
+   * Whether or not the {@link WritableStream} argument should be closed upon
+   * completion.
+   * @default {true}
+   */
+  preventClose?: boolean;
```

Comment:
I think options should be formulated positively rather than negatively, something like `close`, `closeStream` or `closeStreamWhenCompleted`